src/js/components/ValidationProgress.jsx

```diff
@@ -1,61 +1,107 @@
-import React from 'react'
+import React, { Component } from 'react'
 import PropTypes from 'prop-types'
 import * as STATUS from '../constants/statusCodes.js'
 
-export const getFill = props => {
-  let fillWidth = 0
-  let className = 'progressFill'
+let SCALING_FACTOR = 1
+export default class ValidationProgress extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {fillWidth: 0}
+    if(props.file){
+      SCALING_FACTOR = props.file.size/1e6
+      if(SCALING_FACTOR < 1) SCALING_FACTOR = 1
+      if(SCALING_FACTOR > 5) SCALING_FACTOR = 5
+    }
+  }
 
-  if(props.code === STATUS.PARSED_WITH_ERRORS) className += ' error'
+  componentWillReceiveProps(props){
+    if(props.file){
+      SCALING_FACTOR = props.file.size/1e6
+      if(SCALING_FACTOR < 1) SCALING_FACTOR = 1
+      if(SCALING_FACTOR > 5) SCALING_FACTOR = 5
+    }
+  }
 
-  if(props.code <= STATUS.PARSING) fillWidth = props.percentUploaded/3
-  if(props.code >= STATUS.PARSED) fillWidth = 50
-  if(props.code === STATUS.PARSED_WITH_ERRORS ||
-     props.code > STATUS.VALIDATING) fillWidth = 100
+  getText() {
+    let text = 'Uploading...'
+    const code = this.props.code
+    const largeFile = this.props.file && this.props.file.size > 1e5
+    if(code >= STATUS.PARSING) text = 'Analyzing file format...'
+    if(code === STATUS.PARSED_WITH_ERRORS) text = 'File contains formatting errors'
+    if(code === STATUS.VALIDATING) text = 'Validating edits...'
+    if(code > STATUS.VALIDATING) text = 'Edit validation complete'
 
-  return <div className={className} style={{width:fillWidth+'%'}}></div>
-}
+    return (
+      <section className="progressText">
+        <span>{text}</span>
+        {this.getIndicator()}
+        {code === STATUS.VALIDATED_WITH_ERRORS ?
+          <strong>Edits found, review required</strong> :
+          largeFile && code !== STATUS.PARSED_WITH_ERRORS ?
+            <strong>This may take several minutes</strong> :
```

Comment:
@designlanguage seemed to think that it would be helpful to filers to understand that they didn't have to sit and wait and watch. Wouldn't mind hearing from him first. I'm ok either way.